straxen/plugins/acqmon_processing.py

```diff
@@ -0,0 +1,130 @@
+# The start/stop signals for the HEV are generated by the V1495 board
+  
+@export
+@strax.takes_config(
+    strax.Option('max_veto_gap', default = int(5e8),
+                 help = 'Maximum separation between veto stop and start pulses [ns]'),
+    strax.Option('channel_map', track = False, type = immutabledict, 
+                 help = 'immutabledict mapping subdetector to (min, max)'
+                      'channel number.'))
+
+class VetoIntervals(strax.OverlapWindowPlugin):
+    """ Find pairs of veto start and veto stop signals and the veto duration between them
+    busy_*  <= V1495 busy veto for tpc channels
+    he_*    <= V1495 busy veto for high energy tpc channels
+    hev_*   <= DDC10 hardware high energy veto
+    """
+        
+    __version__ ='0.0.24'
+    depends_on = ('aqmon_hits') 
+    provides  = ('veto_intervals')
+    data_kind = ('veto_intervals')
+
+    def infer_dtype(self):
+        dtype = [(('veto start time since unix epoch [ns]', 'time'), np.int64),
+                 (('veto end time since unix epoch [ns]', 'endtime'), np.int64),
+                 (('veto interval [ns]', 'veto_interval'), np.int64),
+                 (('veto signal type', 'veto_type'), np.str('U10'))]
+        return dtype
+    
+    def setup(self):
+        aq_ch_start = straxen.n_hard_aqmon_start
+        aqmon_channel_names = ('sum_wf','m_veto_sync',
+                               'hev_stop', 'hev_start', 'he_stop','he_start','busy_stop', 'busy_start')
+        channel_numbers = np.arange(aq_ch_start, aq_ch_start + len(aqmon_channel_names))
+        
+        # Keeping ch_range as class attribute for now, maybe will need it for other features
+        self.channel_range = self.config['channel_map']['aqmon']
+        self.channel_map = dict(zip(aqmon_channel_names, channel_numbers))
+        self.veto_names = ['busy_','he_','hev_']
+        return self.channel_map
+    
+    def get_window_size(self):
+        # Give a very wide window
+        return (self.config['max_veto_gap']*100)
+    
+    def compute(self, aqmon_hits): 
+        hits = aqmon_hits
+        
+        # Make a dict to populate later
+        res = dict()
+        
+        for i, veto in enumerate(self.veto_names):
+            veto_hits = channel_select(hits, self.channel_map[veto + 'stop'], self.channel_map[veto + 'start'])
+            vetos = self.merge_vetos(veto_hits, gap = self.config['max_veto_gap'],\
+                                           dtype = strax.time_fields, t = 0)
+            
+            # Check that we found a veto interval and update the resulting dict
+            if len(vetos):
+                res.setdefault("time",[]).extend(vetos['time'])
+                res.setdefault("endtime",[]).extend(vetos['endtime'])
+                res.setdefault("veto_interval",[]).extend((vetos['endtime'] - vetos['time']))
+                res.setdefault("veto_type",[]).extend([veto + 'veto']*len(vetos))
+        return res
```

Comment:
I might be wrong, but in case of more than one veto_name in `veto_names`  the result is not sorted by `time` anymore. Could you fix this? I think it is not mandatory by strax, but all our other data_kinds are sorted by time.

Reply:
They are sorted by time for a given channel, so in principle this isn't that different. See:
https://github.com/AxFoundation/strax/blob/master/strax/processing/general.py#L25

Although it doesn't harm to do it here too

Reply:
Hmm...I think they will be sorted though. Physically there can be only a single veto each time, either normal busy, he or hev. So they are bound to come one after another, but maybe I misunderstood your comment.

Reply:
The issue is, the way you are calling your functions in your for loop you will end up with the following result:

```
res = [veto_busy_1, veto_busy_2, veto_busy_3..... veto_busy_n, veto_he_1, veto_he_2, .... veto_he_n, veto_nv_1, veto_nv_2 ...]
```

(symbolically speaking ). Hence your result will be sorted first by channel and then by time, while for all other data_types we sort everything by time and then by channel.

Reply:
Ok, but as far as I know dicts are by their nature unsorted. So any returned dict from a plugin will be unsorted i.e. your comment would apply to stuff like:
https://github.com/XENONnT/straxen/blob/ed27f9c105bcdd9144d754d8d6f31d5841e1c40e/straxen/plugins/peak_processing.py#L257

Reply:
> Ok, but as far as I know dicts are by their nature unsorted. 

You are mixing things up here. Yes the keys with respect to each other are unsorted, but not the values within a key. E.g. 

```
d = dict(a=[1,2,3], v=2)
print(d['a'])
```
will always yield [1,2,3] not [2,1,3] nor [3,2,1].

Reply:
For the record, from the python documentation:
> Changed in version 3.7: Dictionary order is guaranteed to be insertion order. This behavior was an implementation detail of CPython from 3.6.

Reply:
> > Ok, but as far as I know dicts are by their nature unsorted.
> 
> You are mixing things up here. Yes the keys with respect to each other are unsorted, but not the values within a key. E.g.
> 
> ```
> d = dict(a=[1,2,3], v=2)
> print(d['a'])
> ```
> 
> will always yield [1,2,3] not [2,1,3] nor [3,2,1].

Ok, now I understand what you mean. The question I have though, is what does this lack of sorting affects?

Reply:
Currently I think it just confuses the user. But I cannot exclude that their might at some point a check testing if the data is sorted. So it is simply better to be consistent and it is not a big modification.

Reply:
Alright. Then I'll just add:

res_temp = strax.dict_to_rec(res)
res_temp.sort(order = 'time')

To sort out and output the result by time.